src/StyleableBannerWithLink/index.tsx

```diff
@@ -74,14 +82,36 @@ const StyleableBannerWithLink: React.FC<Props> = (props: Props) => {
             buttonText,
             buttonUrl,
             showPaymentIcons = 'false',
+            reminderStage,
+            reminderOption,
             imageUrl,
             imageAltText = '',
             imagePosition = 'center',
+            showPrivacyText = '',
         },
         trackClick,
+        fetchEmail,
     } = props;
 
-    const styles = selfServeStyles(props.brazeMessageProps, defaultColors);
+    const brazeProps = props.brazeMessageProps;
+
+    const styles = selfServeStyles(brazeProps, defaultColors);
+
+    const reminderCtaStyles = {
+        styleReminderButton: brazeProps?.styleReminderButton as OptionalColorValueHex,
+        styleReminderButtonBackground:
+            brazeProps?.styleReminderButtonBackground as OptionalColorValueHex,
+        styleReminderButtonHover: brazeProps?.styleReminderButtonHover as OptionalColorValueHex,
+        styleReminderAnimation: brazeProps?.styleReminderAnimation as OptionalColorValueHex,
```

Comment:
Ah sorry, this doesn't quite address my concern about the casting here. I think we're telling TypeScript something here which isn't necessarily true - that the value from Braze will either be empty (`undefined`) or a valid hex colour code. But I don't think that's correct - it could be anything - so we lose safety which could be provided by TypeScript. We do have a type guard `stringIsColorValueHex` which can safely take us from a `string` to a `ColorValueHex` as long as the value is valid. What happens if you remove the casting here and let the data flow through as a `string | undefined`? I think at some point downstream we validate that the value is a valid hex string, so perhaps that could be the point at which we go from a `string` to a `ColorValueHex` (and use the default if the value provided isn't valid). What do you think?

Reply:
Removing the casting leads to a TS error when I attempt to pass the CTA styles to the CTA component:

<img width="955" alt="Screenshot 2023-11-22 at 09 34 35" src="https://github.com/guardian/braze-components/assets/5357530/fde5343c-94be-48cc-855e-e40a2d3d5800">

Do you want me to spend more time investigating possible alternative solutions to this? Personally, I think in-browser testing all user input for these attributes using `stringIsColorValueHex()` is sufficient to prevent breaking the page CSS in production, and any additional TS won't give us much more beyond code documentation.